packages/dds/sequence/src/test/intervalCollection.undoRedo.spec.ts

```diff
@@ -0,0 +1,165 @@
+import {
+	appendToMergeTreeDeltaRevertibles,
+	MergeTreeDeltaRevertible,
+	revertMergeTreeDeltaRevertibles,
+} from "@fluidframework/merge-tree";
+import { SharedString } from "../sharedString";
+import { IntervalCollection, IntervalType, SequenceInterval } from "../intervalCollection";
+import { SharedStringFactory } from "../sequenceFactory";
+
+const assertIntervals = (
+	sharedString: SharedString,
+	intervalCollection: IntervalCollection<SequenceInterval>,
+	expected: readonly { start: number; end: number }[],
+	validateOverlapping: boolean = true,
+) => {
+	const actual = Array.from(intervalCollection);
+	if (validateOverlapping && sharedString.getLength() > 0) {
+		const overlapping = intervalCollection.findOverlappingIntervals(
+			0,
+			sharedString.getLength() - 1,
+		);
+		assert.deepEqual(actual, overlapping, "Interval search returned inconsistent results");
+	}
+	assert.strictEqual(
+		actual.length,
+		expected.length,
+		`findOverlappingIntervals() must return the expected number of intervals`,
+	);
+
+	const actualPos = actual.map((interval) => {
+		assert(interval);
+		const start = sharedString.localReferencePositionToPosition(interval.start);
+		const end = sharedString.localReferencePositionToPosition(interval.end);
+		return { start, end };
+	});
+	assert.deepEqual(actualPos, expected, "intervals are not as expected");
+};
+
+function assertIntervalEquals(
+	string: SharedString,
+	interval: SequenceInterval | undefined,
+	endpoints: { start: number; end: number },
+): void {
+	assert(interval);
+	assert.equal(
+		string.localReferencePositionToPosition(interval.start),
+		endpoints.start,
+		"mismatched start",
+	);
+	assert.equal(
+		string.localReferencePositionToPosition(interval.end),
+		endpoints.end,
+		"mismatched end",
+	);
+}
+
+describe("Undo/redo for interval collection operations", () => {
+	let sharedString: SharedString;
+	let dataStoreRuntime1: MockFluidDataStoreRuntime;
+	// not sure if i need the factory if im only using one sharedstring
+	let containerRuntimeFactory: MockContainerRuntimeFactory;
+	let collection: IntervalCollection<SequenceInterval>;
+	let revertibles: MergeTreeDeltaRevertible[];
+
+	beforeEach(() => {
+		dataStoreRuntime1 = new MockFluidDataStoreRuntime({ clientId: "1" });
+		sharedString = new SharedString(
+			dataStoreRuntime1,
+			"shared-string-1",
+			SharedStringFactory.Attributes,
+		);
+
+		// unclear if this is needed
+		containerRuntimeFactory = new MockContainerRuntimeFactory();
+
+		// Connect the first SharedString.
+		dataStoreRuntime1.local = false;
+		const containerRuntime1 = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime1);
+		const services1 = {
+			deltaConnection: containerRuntime1.createDeltaConnection(),
+			objectStorage: new MockStorage(),
+		};
+		sharedString.initializeLocal();
+		sharedString.connect(services1);
+		collection = sharedString.getIntervalCollection("test");
+		revertibles = [];
+	});
```

Comment:
You could just match the same setup code as sharedString.spec.ts.

Reply:
I tried this, but the tests were failing on the assertIntervals call with this change. For some reason, it seemed like the intervals weren't being added to the collection, but were being returned from findOverlappignIntervals, so I got the error "Interval search returned inconsistent results".

Reply:
we should really just remove that delay load code, as it doesn't work anymore, and probably should never be used

Reply:
> did you also call `initializeLocal`?

I did, but I'll try the factory create and see if I still get the failures.

Reply:
After playing around with it a bit, I'm still getting failures if I use the stringFactory create instead of the same setup as the intervalCollection tests.

Reply:
Investigation into failures tracked by [AB#4044](https://dev.azure.com/fluidframework/internal/_workitems/edit/4044)